evalml/tests/component_tests/test_featuretools.py

```diff
@@ -56,6 +56,7 @@ def test_featuretools_index(mock_calculate_feature_matrix, mock_dfs, X_y_multi):
     assert arg_tr.to_list() == new_index
 
     # check if _make_entity_set fills in the proper index values
+    feature = DFSTransformer()
```

Comment:
These changes make it so that you cannot take one `DFSTransformer` and fit (generate new features) on new data. I don't believe that we need this use case for the `DFSTransformer` but I'm happy to discuss!

Reply:
Can you explain? Without this line the test still passes

Reply:
@freddyaboulton with the new logic this isn't necessary anymore, will fix!